node/trieIterators/factory/sovereignStakeValuesHandlerFactory.go

```diff
@@ -0,0 +1,24 @@
+package factory
+
+import (
+	"github.com/multiversx/mx-chain-go/node/external"
+	"github.com/multiversx/mx-chain-go/node/trieIterators"
+)
+
+type sovereignTotalStakedValueProcessorFactory struct {
+}
+
+// NewSovereignTotalStakedValueProcessorFactory create a new sovereign total staked value handler
+func NewSovereignTotalStakedValueProcessorFactory() *sovereignTotalStakedValueProcessorFactory {
+	return &sovereignTotalStakedValueProcessorFactory{}
+}
+
+// CreateTotalStakedValueProcessorHandler will create a new instance of total staked value processor for sovereign chain
+func (d *sovereignTotalStakedValueProcessorFactory) CreateTotalStakedValueProcessorHandler(args trieIterators.ArgTrieIteratorProcessor) (external.TotalStakedValueHandler, error) {
```

Comment:
Perhaps rename the pointer receiver to f(from factory?), since `d` is not part of the name of the struct
```suggestion
func (f *sovereignTotalStakedValueProcessorFactory) CreateTotalStakedValueProcessorHandler(args trieIterators.ArgTrieIteratorProcessor) (external.TotalStakedValueHandler, error) {
```